builtin/credential/approle/path_role.go

```diff
@@ -847,6 +823,14 @@ func (b *backend) roleEntry(ctx context.Context, s logical.Storage, roleName str
 		needsUpgrade = true
 	}
 
+	if role.TokenPeriod == 0 && role.Period > 0 {
+		role.TokenPeriod = role.Period
```

Comment:
Can we do `role.Period = 0` in addition to this?

Reply:
We keep it there so we can display it on read, so that we are round-tripping the attribute. Make TF happy, etc.